### PR TITLE
Add a new smoke test: crawling BO

### PR DESCRIPTION
**Pull request by SimonGrn**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add a new smoke test: crawling BO. Also add a few generic methods and some more architecture to work with 
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | npm run specific-test smoke/crawlingBO

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14815)

### CODE REVIEW

File: tests/puppeteer/pages/BO/BO_commonPage.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  /*
+  Methods
+   */
+  /**
+   * Open a subMenu if closed and click on a sublink
+   * @param blockSelector
+   * @param parentSelector
+   * @param linkSelector
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/BO_commonPage.js:37]: Please remove unused  param blockSelector

File: tests/puppeteer/test/smoke/crawlingBO.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  BO_DASHBOARD = await (new BO_DASHBOARD_PAGE(page));
+  BO_COMMON = await (new BO_COMMON_PAGE(page));
+  BO_ORDERS = await (new BO_ORDERS_PAGE(page));
+  BO_PRODUCTS = await (new BO_PRODUCTS_PAGE(page));
+  BO_CUSTOMERS = await (new BO_CUSTOMERS_PAGE(page));
+};
+
+// connect to the BO
+global.scenario('Crawl into BO et check a few key pages', async () => {
+  test('should login into BO', async () => {
```
Comment (boubkerbribri) [inline tests/puppeteer/test/smoke/crawlingBO.js:29]: Please complete description of the scenario in comment

File: tests/puppeteer/pages/BO/BO_commonPage.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  async goToSubMenu(parentSelector, linkSelector) {
+    if (await this.elementVisible(linkSelector)) {
+      await this.page.click(linkSelector);
+    } else {
+      // open the block
+      await this.page.click(parentSelector);
+      await this.page.waitForSelector(`${parentSelector}.open`, {visible: true});
+      await this.page.click(linkSelector);
+    }
+    this.page.waitFor(500);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/BO_commonPage.js:49]: Maybe doing this instead
```js
//Selectors
    this.ordersLink = '#subtab-AdminOrders';
    this.productsLink = '#subtab-AdminProducts';
    this.customersLink = '#subtab-AdminCustomers';

// In goToSubMenu function
    await this.page.waitForSelector(`${linkSelector}.-active`, {visible: true});
```

File: tests/puppeteer/pages/BO/BO_dashboard.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const CommonPage = require('../commonPage');
+
+module.exports = class BO_DASHBOARD extends CommonPage {
+  constructor(page) {
+    super(page);
+
+    this.pageTitle = 'PrestaShop';
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/BO_dashboard.js:7]: Please use `Dashboard *` instead and remove it from BO_commonPage.js

File: tests/puppeteer/pages/BO/BO_commonPage.js (hunk 1 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/BO/BO_commonPage.js:37]: Done thanks

File: tests/puppeteer/test/smoke/crawlingBO.js (hunk 2 above)
Comment (SimonGrn) [reply tests/puppeteer/test/smoke/crawlingBO.js:29]: Done thanks

File: tests/puppeteer/pages/BO/BO_commonPage.js (hunk 3 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/BO/BO_commonPage.js:49]: Doesn't work sadly. I'll try something else.

**Review by matthieu-rolland — APPROVED** [review]
lgtm

File: tests/puppeteer/pages/BO/BO_commonPage.js (hunk 3 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/BO/BO_commonPage.js:49]: It works if I change the Link selector to remove the `>a` part

File: tests/puppeteer/pages/BO/BO_dashboard.js (hunk 4 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/BO/BO_dashboard.js:7]: Done

**Pull request merged**